### DISCUSSION TITLE
Can we use styler/lintr with Quarto yet?

### DISCUSSION THREAD

**Discussion by adasguptaAZ** [quarto-dev/quarto-cli#793] [Q&A] [answered]
Are we able to format and/or lint our R code in Quarto yet? The RStudio styler plug-in is refusing to do so.

**Comment by cderv** [comment]
Hi @adasguptaAZ, 

Is this the error you get ? 
````r
> styler:::style_active_file()
Using style transformers `styler::tidyverse_style()`
Error in `styler:::style_active_file()`:
! Can only style .R, .Rmd and .Rnw files.
````

**styler** needed to support the new `.qmd` extensions and this should be the case in their current development version following issue https://github.com/r-lib/styler/issues/913 and associated PR https://github.com/r-lib/styler/pull/931

You should update **styler** package. 

If you have another issue, this would probably be regarding **styler** - RStudio Addins are extensions that any package author can add. 

Hope it helps

**Comment by mcanouil** [comment, marked as answer]
Quarto documents are supported since styler 1.8.0
